src/api/app/models/cloud/upload_job.rb

```diff
@@ -0,0 +1,40 @@
+module Cloud
+  class UploadJob
+    include ActiveModel::Validations
+    include ActiveModel::Model
+    extend Forwardable
+
+    attr_accessor :user, :params
+    attr_reader :user_upload_job, :backend_upload_job
+    validate :validate_jobs
+    def_delegator :backend_upload_job, :id
+
+    def create
+      self.backend_upload_job = Cloud::Backend::UploadJob.create(user, params)
+      return self unless backend_upload_job.valid?
+
+      self.user_upload_job = @user.upload_jobs.create(
+        job_id: backend_upload_job.id,
+        user: @user
```

Comment:
I don't think you need to explicitly set the user here. It's inherited via the association.

Reply:
Right